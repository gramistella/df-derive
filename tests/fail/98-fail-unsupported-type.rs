use df_derive::ToDataFrame;
use std::collections::HashMap;
#[path = "../common.rs"]
mod core;
use crate::core::dataframe::ToDataFrame;

#[derive(ToDataFrame)]
struct Unsupported {
    id: i32,
    // HashMap is not a supported type for conversion to a Polars Series
    metadata: HashMap<String, String>,
}

fn main() {}
