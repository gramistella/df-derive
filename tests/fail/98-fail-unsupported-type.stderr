error: df-derive does not support HashMap
  --> tests/fail/98-fail-unsupported-type.rs:11:15
   |
11 |     metadata: HashMap<String, String>,
   |               ^^^^^^^^^^^^^^^^^^^^^^^

warning: unused import: `crate::core::dataframe::ToDataFrame`
 --> tests/fail/98-fail-unsupported-type.rs:5:5
  |
5 | use crate::core::dataframe::ToDataFrame;
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default
